bin/cli.rs

```diff
@@ -0,0 +1,126 @@
+#![allow(missing_docs)]
+use std::{
+    collections::BTreeMap,
+    fs::{create_dir_all, File},
+    io::BufReader,
+    num::NonZeroUsize,
+    thread,
+};
+
+use alloy_consensus::constants::KECCAK_EMPTY;
+use alloy_primitives::{Address, B256};
+use alloy_provider::{Provider, ProviderBuilder};
+use alloy_rpc_types::{BlockId, BlockTransactionsKind};
+use clap::{Parser, Subcommand};
+use pevm::{
+    chain::{PevmChain, PevmEthereum},
+    EvmAccount, EvmCode, Pevm, RpcStorage,
+};
+use reqwest::Url;
+use tokio::runtime;
+
+#[derive(Parser, Debug)]
+pub struct Cli {
+    #[clap(subcommand)]
+    pub subcmd: Commands,
+}
+
+#[derive(Subcommand, Debug)]
+pub enum Commands {
+    /// Snapshot a block to disk.
+    Snapshot {
+        /// The ID of the block.
+        block_id: BlockId,
+        /// The RPC url to connect to.
+        rpc_url: String,
+        /// The output directory to write the snapshot to.
```

Comment:
Let's keep it simple by hardcoding `output_dir` and `bytecode_path` to `data/` for now.